water/main.py

```diff
@@ -163,7 +165,13 @@ def update():
         for node1, node2 in G.edges():
             node1_pollution = series[node1]
             node2_pollution = series[node2]
-            edge_values.append((node1_pollution + node2_pollution) / 2.)
+            # The edge color should be the mean of the connected nodes
+            # except when one node is zero, which indicates pollution
+            # is yet to spread through that edge (pipe)
+            if node1_pollution == 0 or node2_pollution == 0:
+                edge_values.append(0)
+            else:
+                edge_values.append((node1_pollution + node2_pollution) / 2.)
```

Comment:
How does this look in motion?

I think having the edge value be the mean helped make the flow appear smoother.

Reply:
it looks the same as before, it's still the mean in most cases, just that the edges where one node has zero pollution are not coloured, which I think makes sense because we can think that pollution is yet to spread through that edge